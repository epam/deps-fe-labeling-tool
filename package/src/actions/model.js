
import { batch } from 'react-redux'
import { createAction } from 'redux-actions'
import { LabelType, LABEL_TYPE_NAME } from '@/models/Label'
import { Markup } from '@/models/Markup'
import { markupSelector } from '@/selectors/markup'
import { fieldsSelector, fieldsToDeleteSelector } from '@/selectors/model'
import {
  updateLabelsWithSettings,
  updateTablesWithSettings
} from './markup'

const FEATURE_NAME = 'MODEL'

const storeFields = createAction(
  `${FEATURE_NAME}/STORE`
)

const storeInitialFields = createAction(
  `${FEATURE_NAME}/STORE_INITIAL_FIELDS`
)

const updateInitialFields = createAction(
  `${FEATURE_NAME}/UPDATE_INITIAL_FIELDS`
)

const enableDeleteFieldsMode = createAction(
  `${FEATURE_NAME}/ENABLE_DELETE_FIELD_MODE`
)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

const updateFieldsToDelete = createAction(
  `${FEATURE_NAME}/UPDATE_FIELDS_TO_DELETE`
)

const cancelDeleteFieldsMode = createAction(
  `${FEATURE_NAME}/CANCEL_DELETE_FIELDS_MODE`
)

const confirmFieldsDeletion = () => (dispatch, getState) => {
  const state = getState()
  const markup = markupSelector(state)
  const fields = fieldsSelector(state)
  const fieldsToDelete = fieldsToDeleteSelector(state)
  const newFields = fields.filter((field) => fieldsToDelete.every((f) => f.code !== field.code))
  const assignedObjects = markup
    ? Markup.getAllObjects(markup).filter((markupObject) => fieldsToDelete.find((field) => field.code === markupObject.fieldCode))
    : []

  batch(() => {
    assignedObjects.forEach((markupObject) => {
      if (markupObject.typeName === LABEL_TYPE_NAME) {
        dispatch(updateLabelsWithSettings({
          ...markupObject,
          fieldCode: '',
          type: LabelType.UNASSIGNED,
          index: undefined
        }))
      } else {
        dispatch(updateTablesWithSettings({
          ...markupObject,
          fieldCode: '',
          index: undefined
        }))
      }
    })
    dispatch(storeFields(newFields))
    dispatch(cancelDeleteFieldsMode())
  })
}

export {
  enableDeleteFieldsMode,
  resetDefault,
  storeFields,
  storeInitialFields,
  updateFieldsToDelete,
  updateInitialFields,
  confirmFieldsDeletion,
  cancelDeleteFieldsMode
}
