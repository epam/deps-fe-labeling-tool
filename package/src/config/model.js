import { FieldType } from '@/enums/FieldType'
import { Field } from '@/models/Field'
import { PairFieldMeta, ListFieldMeta, EnumFieldMeta } from '@/models/FieldMeta'

const fields = [
  new Field(
    'dates',
    'Dates',
    FieldType.DATE
  ),
  new Field(
    'firstName',
    'First Name',
    FieldType.STRING
  ),
  new Field(
    'lastName',
    'Last Name',
    FieldType.STRING,
    undefined,
    true
  ),
  new Field(
    'companyName',
    'Company Name',
    FieldType.STRING,
    undefined,
    true
  ),
  new Field(
    'city',
    'City',
    FieldType.ENUM,
    new EnumFieldMeta([
      'Anytown',
      'Sometown',
      'San Diego',
      'Los Angeles'
    ]),
    true
  ),
  new Field(
    'countryStates',
    'Country States',
    FieldType.LIST,
    new ListFieldMeta(FieldType.STRING),
    true
  ),
  new Field(
    'dateMatching',
    'Date Matching',
    FieldType.LIST,
    new ListFieldMeta(FieldType.CHECKMARK),
    true
  ),
  new Field(
    'date',
    'Date',
    FieldType.PAIR,
    new PairFieldMeta(FieldType.STRING, FieldType.STRING)
  ),
  new Field(
    'zipCodes',
    'Zip Codes',
    FieldType.LIST,
    new ListFieldMeta(
      FieldType.PAIR,
      new PairFieldMeta(FieldType.STRING, FieldType.STRING)
    ),
    true
  ),
  new Field(
    'homeAddress',
    'Home Address',
    FieldType.STRING
  ),
  new Field(
    'businessAddress',
    'Business Address',
    FieldType.PAIR,
    new PairFieldMeta(FieldType.STRING, FieldType.STRING),
    true
  ),
  new Field(
    'taxpayerSpousesInfo',
    'Taxpayer Spouse\'s Info',
    FieldType.TABLE
  ),
  new Field(
    'taxpayerInfo',
    'Taxpayer Info',
    FieldType.TABLE,
    undefined,
    true
  ),
  new Field(
    'tables',
    'Tables',
    FieldType.LIST,
    new ListFieldMeta(FieldType.TABLE)
  ),
  new Field(
    'uniqueIdentifier',
    'Unique Identifier',
    FieldType.PAIR,
    new PairFieldMeta(FieldType.STRING, FieldType.STRING)
  ),
  new Field(
    'signatory',
    'Signatory',
    FieldType.CHECKMARK
  ),
  new Field(
    'returnTranscript',
    'Return Transcript',
    FieldType.CHECKMARK
  ),
  new Field(
    'formNumbers',
    'Form Numbers',
    FieldType.LIST,
    new ListFieldMeta(
      FieldType.ENUM,
      new EnumFieldMeta([
        '1098',
        '5498',
        '1040'
      ])
    )
  )
]

export {
  fields
}
