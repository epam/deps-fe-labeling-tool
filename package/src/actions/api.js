import { createAction } from 'redux-actions'
import { rotationAnglesSelector } from '@/selectors/canvas'
import { pageImageUrlSelector } from '@/selectors/document'
import { markupSelector } from '@/selectors/markup'
import { fieldsSelector } from '@/selectors/model'
import { primaryLanguageSelector } from '@/selectors/ocr'
import { getApi } from '@/services/api'

const FEATURE_NAME = 'API'

const requestAttempt = createAction(
  `${FEATURE_NAME}/REQUEST_ATTEMPT`
)

const requestSuccess = createAction(
  `${FEATURE_NAME}/REQUEST_SUCCESS`
)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

const requestFailure = createAction(
  `${FEATURE_NAME}/REQUEST_FAILURE`,
  (requestId, error) => ({
    requestId,
    error
  })
)

export const createRequestAction = (requestId, actionCreator) => {
  const requestActionCreator = (...args) => async (dispatch, getState) => {
    try {
      dispatch(requestAttempt(requestId))
      const result = await actionCreator(...args)(dispatch, getState)
      dispatch(requestSuccess(requestId))
      return result
    } catch (error) {
      dispatch(requestFailure(requestId, error.message))
      throw error
    }
  }
  requestActionCreator.toString = () => requestId
  return requestActionCreator
}

const save = createRequestAction(
  'save',
  () => (dispatch, getState) => {
    const state = getState()
    const markup = markupSelector(state)
    const rotationAngles = rotationAnglesSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().save(markup, rotationAngles, language)
  }
)

const saveMarkup = createRequestAction(
  'saveMarkup',
  () => (dispatch, getState) => {
    const state = getState()
    const markup = markupSelector(state)
    const rotationAngles = rotationAnglesSelector(state)
    const language = primaryLanguageSelector(state)
    const fields = fieldsSelector(state)
    return getApi().saveMarkup(markup, rotationAngles, language, fields)
  }
)

const ocrText = createRequestAction(
  'ocrText',
  (engine, coordinates) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().ocrText(
      engine,
      pageImageUrl,
      coordinates,
      language
    )
  }
)

const omrArea = createRequestAction(
  'omrArea',
  (coordinates) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)

    return getApi().omrArea(
      pageImageUrl,
      coordinates
    )
  }
)

const detectTables = createRequestAction(
  'detectTables',
  (selectedAreaCoords) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    return getApi().detectTables(
      pageImageUrl,
      selectedAreaCoords
    )
  }
)

const ocrTable = createRequestAction(
  'ocrTable',
  (engine, table) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().ocrTable(
      engine,
      pageImageUrl,
      table,
      language
    )
  }
)

export {
  detectTables,
  ocrTable,
  ocrText,
  omrArea,
  resetDefault,
  requestAttempt,
  requestSuccess,
  requestFailure,
  save,
  saveMarkup
}
