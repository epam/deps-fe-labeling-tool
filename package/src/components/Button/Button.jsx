import React from 'react'
import { StyledButton } from './Button.styles'

const ButtonType = {
  DEFAULT: 'default',
  LINK: 'link',
  PRIMARY: 'primary'
}

const Button = React.forwardRef(
  (props, ref) =>
    <StyledButton
      ref={ref}
      {...props}
    />
)

Button.displayName = 'Button'

export {
  Button,
  ButtonType
}
