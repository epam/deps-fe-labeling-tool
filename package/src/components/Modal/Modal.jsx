import React from 'react'
import { Modal } from 'antd'

const CustomModal = (props) => (
  <Modal {...props} />
)

export {
  CustomModal
}
