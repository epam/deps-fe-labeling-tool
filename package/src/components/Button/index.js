import { Button, ButtonType } from './Button'
import { IconButton } from './IconButton'
import { TextButton } from './TextButton.styles'

Button.Icon = IconButton

export {
  Button,
  ButtonType,
  TextButton
}
