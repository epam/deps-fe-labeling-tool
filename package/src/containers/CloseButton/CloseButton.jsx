import React, { useCallback, useEffect, useMemo, useState } from 'react'
import 'antd/lib/modal/style/index.less'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { save, saveMarkup } from '@/actions/api'
import { ButtonType } from '@/components/Button'
import { CloseIcon } from '@/components/Icons/CloseIcon'
import { LoadingIcon } from '@/components/Icons/LoadingIcon'
import { CustomModal } from '@/components/Modal'
import { Tooltip } from '@/components/Tooltip'
import { fieldShape } from '@/models/Field'
import { ModifiedObjects, modifiedObjectsShape } from '@/models/ModifiedObjects'
import { modifiedObjectsSelector } from '@/selectors/markup'
import {
  fieldsSelector,
  initialFieldsSelector
} from '@/selectors/model'
import { getApi } from '@/services/api'
import { getEvents } from '@/services/events'
import { isEqual } from '@/utils/isEqual'
import {
  Wrapper,
  Button,
  QuestionCircle,
  TitleWrapper,
  CloseOutlined
} from './CloseButton.styles'

const RESOURCE_MODAL_WIDTH = 450
const RESOURCE_TOOLTIP_CLOSE = 'Close'
const RESOURCE_CONFIRM_LEAVE_TITLE = 'Do you want to extract document?'
const RESOURCE_CONFIRM_LEAVE_CONTENT = 'You have unsaved changes. Do you wish to keep them?'
const RESOURCE_WRAPPER_KEY = 'wrapper'
const CANCEL_BUTTON_ID = 'ignore'

const RESOURCE_MODAL_BUTTON_NAME = {
  IGNORE: 'Ignore',
  SAVE: 'Save',
  SAVE_AND_EXTRACT: 'Save & extract'
}

const CloseButton = ({
  fields,
  initialFields,
  modifiedObjects,
  save,
  saveMarkup
}) => {
  const [visible, setVisible] = useState(false)
  const [isSaving, setIsSaving] = useState(false)

  const sameFields = useMemo(
    () => isEqual(fields, initialFields),
    [fields, initialFields]
  )

  const onIgnore = useCallback(
    () => {
      getApi().close()
      setVisible(false)
    }, [])

  const onSaveAndExtract = useCallback(
    async () => {
      try {
        setIsSaving(true)
        await save()
        onIgnore()
      } finally {
        setIsSaving(false)
      }
    }, [onIgnore, save]
  )

  const onSaveMarkup = useCallback(
    async () => {
      try {
        setIsSaving(true)
        await saveMarkup()
        onIgnore()
      } finally {
        setIsSaving(false)
      }
    }, [onIgnore, saveMarkup]
  )

  const onParentCloseEvent = useCallback(() => {
    if (!ModifiedObjects.isModified(modifiedObjects) && sameFields) {
      return false
    }

    setVisible(true)

    return true
  }, [modifiedObjects, sameFields])

  const onClick = useCallback(async () => {
    if (!ModifiedObjects.isModified(modifiedObjects) && sameFields) {
      return getApi().close()
    }

    setVisible(true)
  }, [modifiedObjects, sameFields])

  useEffect(() => {
    const unsubscribe = getEvents().onClose(onParentCloseEvent)

    return () => {
      unsubscribe()
    }
  }, [onParentCloseEvent])

  const isSaveWithExtractionAvailable = () => !!getApi().ocrTable && !!getApi().ocrText

  const renderFooterSaveButtons = useMemo(
    () => {
      if (getApi().save) {
        return (
          <>
            <Button
              type={ButtonType.DEFAULT}
              onClick={onSaveMarkup}
              disabled={isSaving}
            >
              {RESOURCE_MODAL_BUTTON_NAME.SAVE}
            </Button>
            {
              isSaveWithExtractionAvailable() && (
                <Button
                  type={ButtonType.PRIMARY}
                  onClick={onSaveAndExtract}
                  disabled={isSaving}
                >
                  {isSaving && <LoadingIcon />}
                  {RESOURCE_MODAL_BUTTON_NAME.SAVE_AND_EXTRACT}
                </Button>
              )
            }
          </>
        )
      }

      return (
        <Button
          type={ButtonType.PRIMARY}
          onClick={onSaveMarkup}
          disabled={isSaving}
        >
          {isSaving && <LoadingIcon />}
          {RESOURCE_MODAL_BUTTON_NAME.SAVE}
        </Button>
      )
    }, [onSaveMarkup, onSaveAndExtract, isSaving]
  )

  const renderFooter = useMemo(
    () => ([
      <Wrapper key={RESOURCE_WRAPPER_KEY}>
        <Button
          type={ButtonType.LINK}
          onClick={onIgnore}
          disabled={isSaving}
        >
          {RESOURCE_MODAL_BUTTON_NAME.IGNORE}
        </Button>
        {renderFooterSaveButtons}
      </Wrapper>
    ]), [onIgnore, renderFooterSaveButtons, isSaving]
  )

  return (
    <React.Fragment>
      <Tooltip title={RESOURCE_TOOLTIP_CLOSE}>
        <Button.Icon
          icon={<CloseIcon />}
          onClick={onClick}
        />
      </Tooltip>
      <CustomModal
        visible={visible}
        onCancel={() => setVisible(false)}
        cancelButtonProps={{ id: CANCEL_BUTTON_ID }}
        closeIcon={<CloseOutlined />}
        title={
          getApi().saveMarkup && (
            <TitleWrapper>
              <QuestionCircle />
              {RESOURCE_CONFIRM_LEAVE_TITLE}
            </TitleWrapper>
          )
        }
        width={RESOURCE_MODAL_WIDTH}
        closable={true}
        footer={renderFooter}
      >
        {RESOURCE_CONFIRM_LEAVE_CONTENT}
      </CustomModal>
    </React.Fragment>
  )
}

CloseButton.propTypes = {
  fields: PropTypes.arrayOf(fieldShape).isRequired,
  initialFields: PropTypes.arrayOf(fieldShape),
  modifiedObjects: modifiedObjectsShape,
  save: PropTypes.func.isRequired,
  saveMarkup: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  fields: fieldsSelector(state),
  initialFields: initialFieldsSelector(state),
  modifiedObjects: modifiedObjectsSelector(state)
})

const mapDispatchToProps = {
  save,
  saveMarkup
}

const ConnectedComponent = connect(mapStateToProps, mapDispatchToProps)(React.memo(CloseButton))

export {
  ConnectedComponent as CloseButton
}
