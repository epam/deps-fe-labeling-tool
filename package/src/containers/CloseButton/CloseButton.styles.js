import { CloseOutlined, QuestionCircleOutlined } from '@ant-design/icons'
import styled from 'styled-components'
import { Button } from '@/components/Button'
import { COLORS } from '@/theme/theme.default'

const WarningModalContentSpan = styled.span`
  font-weight: 500;
  font-size: 1.7rem;
  padding-bottom: 1.3rem;
`

const WarningModalContent = styled.div`
  padding: 1rem;
`

const StyledButton = styled(Button)`
  font-size: 1.4rem;
  line-height: 2.2rem;
  padding: 0 1.6rem;
`

const Wrapper = styled.div`
  display: flex;

  & > button:nth-child(2) {
    margin-left: auto !important;
  }
`

const TitleWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`

const StyledQuestionCircle = styled(QuestionCircleOutlined)`
  font-size: 2.1rem;
  margin-right: 1.2rem;
  color: ${COLORS.WARNING};
`

const StyledCloseOutlined = styled(CloseOutlined)`
  font-size: 1.4rem;
`

export {
  WarningModalContentSpan,
  WarningModalContent,
  StyledButton as Button,
  StyledQuestionCircle as QuestionCircle,
  StyledCloseOutlined as CloseOutlined,
  Wrapper,
  TitleWrapper
}
