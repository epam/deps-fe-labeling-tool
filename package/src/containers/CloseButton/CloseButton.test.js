import { mockReactRedux } from '@/mocks/mockReactRedux'
import { mockMarkupSelectors } from '@/mocks/selectors/markup'
import { mockModelSelectors } from '@/mocks/selectors/model'
import React from 'react'
import { shallow } from 'enzyme'
import { save, saveMarkup } from '@/actions/api'
import { Button } from '@/components/Button'
import { CustomModal } from '@/components/Modal'
import { CloseButton } from '@/containers/CloseButton/CloseButton'
import { getApi } from '@/services/api'
import { Button as StyledButton } from './CloseButton.styles'

jest.mock('react-redux', () => mockReactRedux)
jest.mock('@/selectors/markup', () => mockMarkupSelectors)
jest.mock('@/selectors/model', () => mockModelSelectors)

const mockApiClose = jest.fn()
const mockOnCloseEvent = jest.fn()
const mockEventTarget = {
  target: {
    innerText: 'Save'
  }
}

jest.mock('@/services/api', () => ({
  getApi: jest.fn(() => ({
    close: mockApiClose,
    save: jest.fn(),
    saveMarkup: jest.fn(),
    ocrText: jest.fn(),
    ocrTable: jest.fn()
  }))
}))

jest.mock('@/services/events', () => ({
  getEvents: () => ({
    onClose: mockOnCloseEvent
  })
}))

jest.mock('@/actions/api', () => ({
  save: jest.fn(),
  saveMarkup: jest.fn()
}))

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn((f) => f())
}))

const { WrappedComponent, mapStateToProps, mapDispatchToProps } = CloseButton

describe('Container: CloseButton', () => {
  describe('mapStateToProps', () => {
    it('should pass correct props to the wrapped component (state)', () => {
      const { props } = mapStateToProps()
      expect(props).toMatchSnapshot()
    })
  })

  describe('mapDispatchToProps', () => {
    it('should pass save action as save prop', () => {
      const { props } = mapDispatchToProps()
      props.save()
      expect(save).toHaveBeenCalledTimes(1)
    })

    it('should pass saveMarkup action as saveMarkup prop', () => {
      const { props } = mapDispatchToProps()
      props.saveMarkup()
      expect(saveMarkup).toHaveBeenCalledTimes(1)
    })
  })

  let defaultProps, wrapper
  beforeEach(() => {
    defaultProps = {
      ...mapStateToProps().props,
      ...mapDispatchToProps().props,
      fieldsToDelete: []
    }

    wrapper = shallow(<WrappedComponent {...defaultProps} />)
  })

  it('should render correct layout', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correct layout if getApi().save is undefined', () => {
    getApi.mockReturnValueOnce({}).mockReturnValueOnce({})
    wrapper = shallow(<WrappedComponent {...defaultProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call apiClose when calling to button onClick', () => {
    const button = wrapper.find(Button.Icon)
    button.simulate('click')
    expect(mockApiClose).toHaveBeenCalled()
  })

  it('should call subscribe on component mount', () => {
    expect(mockOnCloseEvent).toHaveBeenCalled()
  })

  it('should call apiClose when click on button "Ignore"', () => {
    const ModalFooter = wrapper.find(CustomModal).props().footer
    const FooterWrapper = shallow(<div>{ModalFooter}</div>)
    FooterWrapper.find(StyledButton).at(0).props().onClick()
    expect(mockApiClose).toHaveBeenCalled()
  })

  it('should call saveMarkup when click on button "Save"', () => {
    const ModalFooter = wrapper.find(CustomModal).props().footer
    const FooterWrapper = shallow(<div>{ModalFooter}</div>)
    FooterWrapper.find(StyledButton).at(1).props().onClick(mockEventTarget)
    expect(saveMarkup).toHaveBeenCalled()
  })

  it('should call onClose after saving markup', async () => {
    const ModalFooter = wrapper.find(CustomModal).props().footer
    const FooterWrapper = shallow(<div>{ModalFooter}</div>)
    await FooterWrapper.find(StyledButton).at(1).props().onClick(mockEventTarget)
    expect(mockApiClose).toHaveBeenCalled()
  })

  it('should call save when click on button "Save & extract"', () => {
    mockEventTarget.target.innerText = 'Save & extract'
    const ModalFooter = wrapper.find(CustomModal).props().footer
    const FooterWrapper = shallow(<div>{ModalFooter}</div>)
    FooterWrapper.find(StyledButton).at(2).props().onClick(mockEventTarget)
    expect(save).toHaveBeenCalled()
  })

  it('should call onClose after saving and extracting is finished', async () => {
    mockEventTarget.target.innerText = 'Save & extract'
    const ModalFooter = wrapper.find(CustomModal).props().footer
    const FooterWrapper = shallow(<div>{ModalFooter}</div>)
    FooterWrapper.find(StyledButton).at(2).props().onClick(mockEventTarget)
    await save()
    expect(mockApiClose).toHaveBeenCalled()
  })
})
