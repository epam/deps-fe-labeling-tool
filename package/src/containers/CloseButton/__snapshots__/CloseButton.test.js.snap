// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: CloseButton mapStateToProps should pass correct props to the wrapped component (state) 1`] = `
Object {
  "fields": Array [
    Field {
      "code": "123",
      "fieldType": "string",
      "name": "TestField",
      "required": false,
    },
    Field {
      "code": "enum",
      "fieldMeta": Object {
        "options": Array [
          "test",
          "test1",
        ],
      },
      "fieldType": "enum",
      "name": "TestField",
      "required": false,
    },
    Field {
      "code": "enumList",
      "fieldMeta": ListFieldMeta {
        "baseType": "enum",
        "baseTypeMeta": EnumFieldMeta {
          "options": Array [
            "test",
            "test1",
          ],
        },
      },
      "fieldType": "list",
      "name": "EnumerationList",
      "required": false,
    },
  ],
  "initialFields": Array [
    Field {
      "code": "123",
      "fieldType": "string",
      "name": "TestField",
      "required": false,
    },
    Field {
      "code": "enum",
      "fieldMeta": Object {
        "options": Array [
          "test",
          "test1",
        ],
      },
      "fieldType": "enum",
      "name": "TestField",
      "required": false,
    },
    Field {
      "code": "enumList",
      "fieldMeta": ListFieldMeta {
        "baseType": "enum",
        "baseTypeMeta": EnumFieldMeta {
          "options": Array [
            "test",
            "test1",
          ],
        },
      },
      "fieldType": "list",
      "name": "EnumerationList",
      "required": false,
    },
  ],
  "modifiedObjects": Array [],
}
`;

exports[`Container: CloseButton should render correct layout 1`] = `
<Fragment>
  <Tooltip
    title="Close"
  >
    <IconButton
      icon={<CloseIcon />}
      onClick={[Function]}
    />
  </Tooltip>
  <CustomModal
    cancelButtonProps={
      Object {
        "id": "ignore",
      }
    }
    closable={true}
    closeIcon={<CloseButtonstyles__StyledCloseOutlined />}
    footer={
      Array [
        <CloseButtonstyles__Wrapper>
          <CloseButtonstyles__StyledButton
            disabled={false}
            onClick={[Function]}
            type="link"
          >
            Ignore
          </CloseButtonstyles__StyledButton>
          <React.Fragment>
            <CloseButtonstyles__StyledButton
              disabled={false}
              onClick={[Function]}
              type="default"
            >
              Save
            </CloseButtonstyles__StyledButton>
            <CloseButtonstyles__StyledButton
              disabled={false}
              onClick={[Function]}
              type="primary"
            >
              Save & extract
            </CloseButtonstyles__StyledButton>
          </React.Fragment>
        </CloseButtonstyles__Wrapper>,
      ]
    }
    onCancel={[Function]}
    title={
      <CloseButtonstyles__TitleWrapper>
        <CloseButtonstyles__StyledQuestionCircle />
        Do you want to extract document?
      </CloseButtonstyles__TitleWrapper>
    }
    visible={false}
    width={450}
  >
    You have unsaved changes. Do you wish to keep them?
  </CustomModal>
</Fragment>
`;

exports[`Container: CloseButton should render correct layout if getApi().save is undefined 1`] = `
<Fragment>
  <Tooltip
    title="Close"
  >
    <IconButton
      icon={<CloseIcon />}
      onClick={[Function]}
    />
  </Tooltip>
  <CustomModal
    cancelButtonProps={
      Object {
        "id": "ignore",
      }
    }
    closable={true}
    closeIcon={<CloseButtonstyles__StyledCloseOutlined />}
    footer={
      Array [
        <CloseButtonstyles__Wrapper>
          <CloseButtonstyles__StyledButton
            disabled={false}
            onClick={[Function]}
            type="link"
          >
            Ignore
          </CloseButtonstyles__StyledButton>
          <CloseButtonstyles__StyledButton
            disabled={false}
            onClick={[Function]}
            type="primary"
          >
            Save
          </CloseButtonstyles__StyledButton>
        </CloseButtonstyles__Wrapper>,
      ]
    }
    onCancel={[Function]}
    visible={false}
    width={450}
  >
    You have unsaved changes. Do you wish to keep them?
  </CustomModal>
</Fragment>
`;
